workflows/differentialabundance.nf

```diff
@@ -66,12 +69,21 @@ if (params.study_type == 'affy_array'){
 // Check optional parameters
 if (params.transcript_length_matrix) { ch_transcript_lengths = Channel.of([ exp_meta, file(params.transcript_length_matrix, checkIfExists: true)]).first() } else { ch_transcript_lengths = [[],[]] }
 if (params.control_features) { ch_control_features = Channel.of([ exp_meta, file(params.control_features, checkIfExists: true)]).first() } else { ch_control_features = [[],[]] }
-if (params.gsea_run) {
-    if (params.gsea_gene_sets){
-        gene_sets_files = params.gsea_gene_sets.split(",")
+
+def run_gene_set_analysis = params.gsea_run || params.gprofiler2_run
+
+if (run_gene_set_analysis) {
```

Comment:
If you really don't think you can handle running the gprofiler module multiple times for different gene set files, suggest modifying the validation logic like:

```
if (run_gene_set_analysis) {
    if (params.gene_sets_files) {
        gene_sets_files = params.gene_sets_files.split(",")
        ch_gene_sets = Channel.of(gene_sets_files).map { file(it, checkIfExists: true) }
        if (params.gprofiler2_run && (!params.gprofiler2_token && !params.gprofiler2_organism) && gene_sets_files.size() > 1) {
            error("gprofiler2 can only work with a single gene set file")
        }
    } else if (params.gsea_run) {
        error("GSEA activated but gene set file not specified!")
    } else if (params.gprofiler2_run) {
        if (!params.gprofiler2_token && !params.gprofiler2_organism) {
            error("To run gprofiler2, please provide a run token, GMT file or organism!")
        }
    } else {
        ch_gene_sets = []    // For methods that can run without gene sets
    }
}
```

(sorry for not making this a proper suggestion- GitHub wouldn't let me becuase of the deleted lines)

Reply:
Ah yes, that is simpler of course.

No, I will certainly enable multiple GMTs in the future when I get around to it, just not right now (I'll mention that in the error).